packages/odyssey-react/src/components/Text/Text.tsx

```diff
@@ -105,42 +120,14 @@ export interface TextProps {
    * The overflow wrapping behavior for the text content.
    * @default normal
    */
-  wrap?: "normal" | "breakWord" | "anywhere" | "nowrap";
-
-  /**
-   * The text color style on hover.
-   */
-  hoverColor?: "body" | "bodyInverse" | "primaryDark";
-
-  /**
-   * The text color style on focus.
-   */
-  focusColor?: "body" | "bodyInverse" | "primary";
-
-  /**
-   * The text color style when disabled.
-   */
-  disabledColor?: "bodyInverse" | "primaryLight" | "sub";
-
-  /**
-   * The text color style when the parent element is hovered.
-   */
-  parentHoveredColor?: "body" | "bodyInverse" | "primaryDark";
-
-  /**
-   * The text color style when the parent element has focus.
-   */
-  parentFocusedColor?: "body" | "bodyInverse" | "primary";
-
-  /**
-   * The text color style when the parent element is disabled.
-   */
-  parentDisabledColor?: "bodyInverse" | "primaryLight" | "sub";
+  overflowWrap?: "normal" | "break-word" | "anywhere";
+}
 
-  /**
-   * Transitions
-   */
-  transition?: "inherit";
+function convertProp(prop: string) {
+  function clearAndUpper(text: string) {
+    return text.replace(/-/, "").toUpperCase();
+  }
+  return prop.replace(/-\w/g, clearAndUpper);
```

Comment:
Three things - first, can you remind me why we're changing our prop arguments to use hyphen-case-like-this now? I'm of the mind that this type of extra transform code and processing props on every render isn't a huge deal but it also seems unideal. Are we trying to make this feel exactly like HTML + CSS? There are already parts of JSX that are very much not HTML + CSS so this seems like a low value change.

Second, if we want to keep this approach for some other reason I haven't considered I'd love to see it extracted as a utility and add a quick unit if we're going to share this around other components. I'm fairly sure I've seen this defined somewhere else in another module already?

Third, the structure of this code is a bit funny, why the separate named function declaration inside the body of `covertProp` if there is no closure or dynamic args happening? It would be simpler, and the code more clear to have the replace callback defined elsewhere or just inlined.

Reply:
"There are already parts of JSX that are very much not HTML + CSS so this seems like a low value change."
I disagree, the parts where JSX diverges from the standard are all driven by language constraints in JavaScript. Keeping the standard hyphenated values has three main benefits
1) transparency, the value maps directly to its DOM equivalent 
2) leveraging previous developer knowledge. No new similar ( but different ) values to learn or be confused about
3) searchable and self documenting nature of those values when developers have questions about their effect

I'm fine externalizing this code to utils

Reply:
@andrewberg-okta thanks I'm aware of those benefits and they do have value! I suppose I don't find asking folks to use camelCase within a React/JS/JSX environment to be a bridge too far but also don't feel incredibly strongly. 

Eventually we should try to standardize even non HTML props: e.g. `variant="invertedSecondary"` vs `variant="inverted-secondary"`

Externalizing to a util would be good for long term maintenance thank you! Lets get this PR merged!